libs/remote_storage/src/s3_bucket.rs

```diff
@@ -332,6 +332,53 @@ impl RemoteStorage for S3Bucket {
         Ok(document_keys)
     }
 
+    async fn list_files(
+        &self,
+        folder: Option<&RemotePath>
+    ) -> anyhow::Result<Vec<RemotePath>>{
+        // TODO: should we use DownloadError error type instead of anyhow::Error?
```

Comment:
should we do the path append `/` thing here?

Reply:
For me, if you put a `/` then it doesn't work.